src/encoder.rs

```diff
@@ -199,9 +199,9 @@ impl<W: Write> Encoder<W> {
 }
 
 /// PNG writer
-pub struct Writer<W: Write> {
+pub struct Writer<'a, W: Write> {
     w: W,
-    info: Info,
+    info: Info<'a>,
```

Comment:
Are the relevant fields actually used, from here one forward? If not then it would also be possible to remove the lifetime parameter here (thus addressing the concern of `StreamWriter`). We'd need to rebuild a new `Info<'static>` from the non-borrowed parts of the struct built in `Encoder`, and drop all borrowed parts.

Reply:
The `Writer` has a `init()` method (used by the `write_header` implementation of `Encoder`) that calls `encode` in `Info<'a>`. In turn, `encode` writes both PLTE and tRNS chunks if present. I'll think more about how can I tackle the `StreamWriter` concern with this information, but I'd say that `Writer` indirectly uses those fields.